README.md

```diff
@@ -275,6 +275,126 @@ An example policy would look like;
 }
 ```
 
+## Message formats and AWS interactions
+
+JustSaying uses a number of Amazon Web Services APIs to transport messages between publishers and subscribers. Since these HTTP APIs can be used by applications which do not use JustSaying, it is entirely possible to have, for example, a Java application publishing a message which is subscribed to by an application using JustSaying. Equally, a C# application publishing messages using JustSaying, which are subscribed to by a Java application, is fully supported by AWS. In order to support this interoperability, it is important that the actual message formats are described. Since Amazon provide SDKs for many programming languages and frameworks, you are unlikely to interact with the APIs purely via HTTP, but knowing the structure of the JustSaying message JSON itself is useful for cross-language purposes.
+
+### Publishing to SNS
+
+JustSaying uses the SNS Publish API to send messages to SNS (documented [here](https://docs.aws.amazon.com/sns/latest/api/API_Publish.html)). The following parameters are set:
+
+#### TopicArn
+The ARN of the topic to which the message is being published.
+
+#### Subject
+JustSaying will set this to the type of the .NET message class - "OrderAccepted", for example.
```

Comment:
_"...to the type name of the .NET message class..."_?

Reply:
`OrderAcceptedMessage` for the type might be a little bit more explicit/obvious

Reply:
Although we do this, there is no need to enforce this as a constraint. In fact, I suspect it would be better not to. I think we would be better to suggest that the Subject should either be an Imperative if it is a Command i.e. DoCalculation and an Event a Past Participle CalculationDone.

It would then be up to consuming frameworks to understand how to map this to a type that they could dispatch.

We could mention that by convention .NET clients just use the message class

Reply:
If you (in Python, say) publish a message to SNS that you want JustSaying to be capable of subscribing to, then you need to know that JustSaying uses the type name here because AWS sends that string to SQS where it's visible from the ReceiveMessage API in Messages ➡️ Message ➡️ Body ➡️ Subject and the JustSaying subscriber then assumes that string is a type name for deserialisation purposes. The constraints for allowed values in the `Subject` property are much more permissive than those for C# identifiers, so if you don't follow the C# rules you can easily publish messages that JustSaying will be unable to subscribe to (without replacing the serialisation implementation (which you can do...)).

Reply:
I've added a comment about the restrictions on the Subject parameter:

> Since a JustSaying _subscriber_ relies on this string to know what type to deserialise the JSON in the Message parameter into, non-JustSaying _publishers_ should restrict this parameter to alphanumeric characters.